static/src/javascripts/projects/commercial/modules/user-features.js

```diff
@@ -0,0 +1,94 @@
+// @flow
+import { getCookie, removeCookie, addCookie } from 'lib/cookies';
+import config from 'lib/config';
+import fetchJson from 'lib/fetch-json';
+import identity from 'common/modules/identity/api';
+// Persistence keys
+const USER_FEATURES_EXPIRY_COOKIE = 'gu_user_features_expiry';
+const PAYING_MEMBER_COOKIE = 'gu_paying_member';
+const AD_FREE_USER_COOKIE = 'GU_AFU';
+
+const userHasData = (): ?string =>
+    getCookie(USER_FEATURES_EXPIRY_COOKIE) ||
+    getCookie(PAYING_MEMBER_COOKIE) ||
+    getCookie(AD_FREE_USER_COOKIE);
+
+const persistResponse = (JsonResponse: () => void) => {
+    const expiryDate = new Date();
+    expiryDate.setDate(expiryDate.getDate() + 1);
+    addCookie(USER_FEATURES_EXPIRY_COOKIE, expiryDate.getTime().toString());
+    addCookie(PAYING_MEMBER_COOKIE, !JsonResponse.adblockMessage);
+    addCookie(AD_FREE_USER_COOKIE, JsonResponse.adFree);
+};
+
+const deleteOldData = (): void => {
+    // We expect adfree cookies to be cleaned up by the logout process, but what if the user's login simply times out?
+    removeCookie(USER_FEATURES_EXPIRY_COOKIE);
+    removeCookie(PAYING_MEMBER_COOKIE);
+    removeCookie(AD_FREE_USER_COOKIE);
+};
+
+const requestNewData = (): void => {
+    fetchJson(`${config.page.userAttributesApiUrl}/me/features`, {
+        mode: 'cors',
+        credentials: 'include',
+    })
+        .then(persistResponse)
+        .catch(() => {});
+};
+
+const featuresDataIsMissing = (): boolean =>
+    !getCookie(USER_FEATURES_EXPIRY_COOKIE) || !getCookie(AD_FREE_USER_COOKIE);
+
+const featuresDataIsOld = (): boolean => {
+    const featuresExpiryCookie = getCookie(USER_FEATURES_EXPIRY_COOKIE);
+    const featuresExpiryTime = parseInt(featuresExpiryCookie, 10);
+    const timeNow = new Date().getTime();
+    return timeNow >= featuresExpiryTime;
+};
+
+const userNeedsNewFeatureData = (): boolean =>
+    featuresDataIsMissing() || featuresDataIsOld();
+
+const userHasDataAfterSignout = (): boolean =>
+    !identity.isUserLoggedIn() && !!userHasData();
+
+/**
+     * Updates the user's data in a lazy fashion
+     */
+
+const refresh = (): void => {
+    if (identity.isUserLoggedIn() && userNeedsNewFeatureData()) {
+        // eslint-disable-next-line no-use-before-define
+        _.requestNewData();
```

Comment:
Does this need the underscore here? The underscore convention is to expose module functions for testing... Not sure if platform are doing it yet, but IIRC, there was talk of using tree-shaking to remove the `export _ = { ...` from PROD bundles

Within the module, we can use the functions without having to reference the exported object ( `_` ):

`_.requestNewData();` >> `requestNewData();`